azkaban-exec-server/src/main/java/azkaban/execapp/JobRunner.java

```diff
@@ -210,38 +217,118 @@ private void createLogger() {
               + this.jobId;
       logger = Logger.getLogger(loggerName);
 
-      // Create file appender
-      String logName = createLogFileName(node);
-      logFile = new File(workingDir, logName);
-
-      String absolutePath = logFile.getAbsolutePath();
-
-      jobAppender = null;
+      // Attempt to create appenders as necessary
       try {
-        RollingFileAppender fileAppender =
-            new RollingFileAppender(loggerLayout, absolutePath, true);
-        fileAppender.setMaxBackupIndex(jobLogBackupIndex);
-        fileAppender.setMaxFileSize(jobLogChunkSize);
-        jobAppender = fileAppender;
-        logger.addAppender(jobAppender);
-        logger.setAdditivity(false);
+        attachFileAppender(createFileAppender());
       } catch (IOException e) {
+        removeAppender(jobAppender);
         flowLogger.error("Could not open log file in " + workingDir
             + " for job " + this.jobId, e);
       }
+
+      AzkabanExecutorServer exec = AzkabanExecutorServer.getApp();
+      if (exec != null && !exec.getAzkabanProps().getBoolean("kafkaLogging.globalDisable", false)
+          && props.getBoolean("azkaban.job.kafkaLogging.enabled", false)) {
```

Comment:
What happens if the kafka server config is not configured and user enables this feature at the job level?